src/elm/Pile.elm

```diff
@@ -0,0 +1,240 @@
+    List.foldl addCard emptyPile elements
+
+
+allCards : CardPile -> List Card
+allCards { activeCard, idleCards } =
+    activeCard
+        |> Maybe.map (\c -> c :: idleCards)
+        |> Maybe.withDefault idleCards
+
+
+startDragging : Id -> CardPile -> CardPile
+startDragging id pile =
+    let
+        ( targetAsList, others ) =
+            List.partition (\card -> card.id == id) pile.idleCards
+    in
+    { pile
+        | idleCards = others
+        , activeCard = targetAsList |> List.head
+    }
+
+
+stopDragging : CardPile -> CardPile
+stopDragging pack =
+    { pack | idleCards = allCards pack, activeCard = Nothing }
+
+
+dragActiveBy : Math.Vector2.Vec2 -> CardPile -> CardPile
+dragActiveBy delta ({ activeCard } as pack) =
+    { pack | activeCard = activeCard |> Maybe.map (dragCardBy delta) }
+
+
+type alias Model =
+    { activePile : Maybe Data.SectionId
+    , piles : AssocList.Dict Data.SectionId CardPile
+    , drag : Draggable.State DragState
+    }
+
+
+type alias DragState =
+    { dictKey : Data.SectionId
+    , cardId : Id
+    }
+
+
+type Msg
+    = DragMsg (Draggable.Msg DragState)
+    | OnDragBy Math.Vector2.Vec2
+    | StartDragging DragState
+    | StopDragging
+
+
+init : List ( Data.SectionId, List Data ) -> Model
+init prePiles =
+    -- will probably either radomly generate these vectors within bounds or
+    -- have some hand crafted numbers for initial layout
+    prePiles
+        |> List.map
+            (\( key, cardsContent ) ->
+                let
+                    xs =
+                        List.range 0 (List.length cardsContent)
+                            |> List.map
+                                (\number ->
+                                    if modBy 2 number == 0 then
+                                        number * 120
+
+                                    else
+                                        number * 85
+                                )
+                            |> List.map toFloat
+
+                    ys =
+                        List.range 0 (List.length cardsContent)
+                            |> List.map
+                                (\number ->
+                                    if modBy 2 number == 0 then
+                                        number * 60
+
+                                    else
+                                        number * 75
+                                )
+                            |> List.map toFloat
+                in
+                cardsContent
+                    |> List.map3 (\x y cont -> ( Math.Vector2.vec2 x y, cont )) xs ys
+                    |> makeCardPile
+                    |> (\newPile -> ( key, newPile ))
+            )
+        |> AssocList.fromList
+        |> (\pileDict -> Model Nothing pileDict Draggable.init)
+
+
+
+-- These functions require wrapping in the programs main Msg type
+-- in order to be compatible
+
+
+dragConfig : (Msg -> msg) -> Draggable.Config DragState msg
```

Comment:
Can't quite recognise this pattern. Usually I would `Html.map` an (SubMsg -> MainAppMsg) as part of the View code so the can't grok this as a config, but I am sure it works and is fine!

Reply:
I looked into this more and I could the message map thing on the `Cmd` and the `Sub` so I've refactored to make it more consistent with how we use the `Html`